src/Symfony/Bridge/Twig/Extension/HttpFoundationExtension.php

```diff
@@ -24,13 +25,27 @@
  */
 class HttpFoundationExtension extends AbstractExtension
 {
-    private $requestStack;
-    private $requestContext;
+    private $urlHelper;
 
-    public function __construct(RequestStack $requestStack, RequestContext $requestContext = null)
+    public function __construct($urlHelper, RequestContext $requestContext = null)
```

Comment:
I think we should add a doc block documenting the correct type of `$urlHelper`. Also, I would remove `$requestContext` from the signature to show that the user is not supposed to pass it anymore. You can still fetch it with `func_get_arg()` for the BC layer.